db/migrations/V41__add_company_roles.sql

```diff
@@ -0,0 +1,8 @@
+-- # adding company roles to company
+ALTER TABLE company
+    ADD COLUMN IF NOT EXISTS roles TEXT[] DEFAULT '{"operator"}';
+
+-- # add authority role to existing authorities
+UPDATE company SET roles = array_append(roles, 'authority') WHERE id IN (
+    SELECT DISTINCT partner_a_id FROM partnership WHERE partnership.type = 'authority-provider'
+    ) AND NOT company.roles @> ARRAY['authority'];
```

Comment:
Better formatting:
```suggestion
-- # add authority role to existing authorities
UPDATE company
   SET roles = ARRAY_APPEND(roles, 'authority')
 WHERE id IN (SELECT DISTINCT partner_a_id
                FROM partnership
               WHERE partnership.type = 'authority-provider')
   AND NOT company.roles @> ARRAY ['authority'];
```

I'm rather pedantic with SQL formatting. The way I do this is that I use IDEA's console and simply hit Command+Option+L (or Reformat Code action from the menus if that's your thing), adding newlines so that each statement is on its own row. The curious right alignment after first word has a bit of a silly history if you haven't met it, but that's basically how SQL was originally written in time of pen and paper, before monitors became a thing. This practice was also highly compatible with patching, which also is an ancient computing term, which is literally cutting pieces of other code snippets and pasting them on top of other pieces, again, still on paper, although the term itself comes from fixing punch cards, not written out statements.

Anyway. Better formatting, s'il vous plaît :)